messybrainz/fetch_and_store_artist_mbids.py

```diff
@@ -0,0 +1,92 @@
+# Script to fetch artist MBIDs from MusicBrainz Database using
+# the recording MBIDs in the MessyBrainz database.
+
+import brainzutils.musicbrainz_db.recording as mb_recording
+import json
+
+from brainzutils import musicbrainz_db
+from brainzutils.musicbrainz_db.exceptions import NoDataFoundException
+from messybrainz import db
+from messybrainz import data
+from sqlalchemy import text
+
+
+def insert_artist_mbids(connection, recording_mbid, artist_mbids):
+    """ Inserts the artist_mbids corresponding to the recording_mbids
+        into the recording_artist_join table.
+    """
+
+    query = text("""INSERT INTO recording_artist_join (recording_mbid, artist_mbid, updated)
+                         VALUES (:recording_mbid, :artist_mbid, now())""")
+
+    values = [
+        {"recording_mbid": recording_mbid, "artist_mbid": artist_mbid} for artist_mbid in artist_mbids
+    ]
+
+    connection.execute(query, values)
+
+
+def fetch_and_store_artist_mbids(connection, recording_mbid):
+    """ Fetches artist MBIDs from the MusicBrainz database for the recording MBID.
+        And inserts the artist MBIDs into the recording_artist_join table.
+    """
+
+    recording = mb_recording.get_recording_by_mbid(recording_mbid, includes=['artists'])
+
+    artist_mbids = []
+    for artist in recording['artists']:
+        artist_mbids.append(artist['id'])
+
+    insert_artist_mbids(connection, recording_mbid, artist_mbids)
+
+
+def fetch_recording_mbids_not_in_recording_artist_join(connection):
+    """ Fetches recording MBIDs that are present in recording_json table
+        but are not present in recording_artist_join table and returns
+        those recording MBIDs.
+    """
+
+    query = text("""SELECT DISTINCT rj.data ->> 'recording_mbid'
+                               FROM recording_json AS rj
+                          LEFT JOIN recording_artist_join AS raj
+                                 ON (rj.data ->> 'recording_mbid')::uuid = raj.recording_mbid
+                              WHERE rj.data ->> 'recording_mbid' IS NOT NULL
+                                AND raj.recording_mbid IS NULL
+    """)
+
+    result = connection.execute(query)
+
+    return result
```

Comment:
Note that directly returning `result` here returns an sqlalchemy ResultProxy element. Maybe we should do a fetchall and only return a list of uuids?